docs/sql-ref-syntax-qry-select-transform.md

```diff
@@ -0,0 +1,262 @@
+clause to embed the mapper or the reducer scripts.
+
+### Syntax
+
+```sql
+SELECT { TRANSFORM ( named_expression [ , ... ] ) | MAP named_expression [ , ... ] | REDUCE named_expression [ , ... ] }
+    [ rowFormat ]
+    [ RECORDWRITER recordWriter_class ]
+    USING script [ AS ( [ col_name [ col_type ] ] [ , ... ] ) ]
+    [ rowFormat ]
+    [ RECORDREADER recordReader_class ]
+```
+
+While `rowFormat` are defined as
+```sql
+{ ROW FORMAT SERDE serde_class [ WITH SERDEPROPERTIES serde_props ] | 
+    ROW FORMAT DELIMITED
+        [ FIELDS TERMINATED BY fields_terminated_char [ ESCAPED BY escapedBy ] ]
+        [ COLLECTION ITEMS TERMINATED BY collectionItemsTerminatedBy ]
+        [ MAP KEYS TERMINATED BY keysTerminatedBy ]
+        [ LINES TERMINATED BY linesSeparatedBy ]
+        [ NULL DEFINED AS nullDefinedAs ] }
+```
+
+### Parameters
+
+* **named_expression**
+
+    An expression with an assigned name. In general, it denotes a column expression.
+
+    **Syntax:** `expression [AS] [alias]`
+
+* **row_format**    
+
+    Spark uses the `SERDE` clause to specify a custom SerDe for one table. Otherwise, use the `DELIMITED` clause to use the native SerDe and specify the delimiter, escape character, null character and so on.
+
+* **SERDE**
+
+    Specifies a custom SerDe for one table.
+
+* **serde_class**
+
+    Specifies a fully-qualified class name of a custom SerDe.
+
+* **SERDEPROPERTIES**
+
+    A list of key-value pairs that is used to tag the SerDe definition.
+
+* **DELIMITED**
+
+    The `DELIMITED` clause can be used to specify the native SerDe and state the delimiter, escape character, null character and so on.
+    
+* **FIELDS TERMINATED BY**
+
+    Used to define a column separator.
+    
+* **COLLECTION ITEMS TERMINATED BY**
+
+    Used to define a collection item separator.
+   
+* **MAP KEYS TERMINATED BY**
+
+    Used to define a map key separator.
+    
+* **LINES TERMINATED BY**
+
+    Used to define a row separator.
+    
+* **NULL DEFINED AS**
+
+    Used to define the specific value for NULL.
+    
+* **ESCAPED BY**
+
+    Used for escape mechanism.
+
+* **RECORDREADER**
+
+    Specifies a custom RecordReader for one table.
+
+* **RECORDWRITER**
+
+    Specifies a custom RecordWriter for one table.
+
+* **recordReader_class**
+
+    Specifies a fully-qualified class name of a custom RecordReader. A default value is `org.apache.hadoop.hive.ql.exec.TextRecordReader`.
+
+* **recordWriter_class**
+
+    Specifies a fully-qualified class name of a custom RecordWriter. A default value is `org.apache.hadoop.hive.ql.exec.TextRecordWriter`.
+
+* **script**
+
+    Specifies a command to process data.
+
+### Serde behavior
+
+Spark uses Hive Serde `org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe` by default, columns will be transformed
+to `STRING` and delimited by tabs before feeding to the user script. Similarly, all `NULL` values will be converted
```

Comment:
`delimited by tabs` -> `combined by tags`? The subject is "columns"

Reply:
nit: Remove `Similarly`, as the NULL handling is not similar to "combine by tabs" at all.

Reply:
Done